server/elasticsearch/src/test/java/org/hawkular/apm/server/elasticsearch/AnalyticsServiceElasticsearchTest.java

```diff
@@ -2212,6 +2231,187 @@ public void testGetCommunicationSummaryStatisticsWithoutOps() {
+        ct2_3.setDuration(600);
+        cts.add(ct2_3);
+
+        try {
+            analytics.storeFragmentCompletionTimes(null, cts);
+        } catch (Exception e) {
+            fail("Failed to store: " + e);
+        }
+
+        CommunicationDetails cd1 = new CommunicationDetails();
+        cd1.setLinkId("cd1");
+        cd1.setBusinessTransaction("testapp");
+        cd1.setTimestamp(1500);
+        cd1.setLatency(100);
+        cd1.setSource(EndpointUtil.encodeEndpoint(null, "in1"));
+        cd1.setTarget(EndpointUtil.encodeEndpoint(null, "out1.1"));
+        cds.add(cd1);
+
+        CommunicationDetails cd2 = new CommunicationDetails();
+        cd2.setLinkId("cd2");
+        cd2.setBusinessTransaction("testapp");
+        cd2.setTimestamp(1500);
+        cd2.setLatency(200);
+        cd2.setSource(EndpointUtil.encodeEndpoint(null, "in1"));
+        cd2.setTarget(EndpointUtil.encodeEndpoint(null, "out1.2"));
+        cds.add(cd2);
+
+        CommunicationDetails cd3 = new CommunicationDetails();
+        cd3.setLinkId("cd3");
+        cd3.setBusinessTransaction("testapp");
+        cd3.setTimestamp(1500);
+        cd3.setLatency(300);
+        cd3.setSource(EndpointUtil.encodeEndpoint(null, "in2"));
+        cd3.setTarget(EndpointUtil.encodeEndpoint(null, "out2.1"));
+        cds.add(cd3);
+
+        CommunicationDetails cd4 = new CommunicationDetails();
+        cd4.setLinkId("cd4");
+        cd4.setBusinessTransaction("testapp");
+        cd4.setTimestamp(1600);
+        cd4.setLatency(300);
+        cd4.setSource(EndpointUtil.encodeEndpoint(null, "in1"));
+        cd4.setTarget(EndpointUtil.encodeEndpoint(null, "out1.2"));
+        cds.add(cd4);
+
+        CommunicationDetails cd5 = new CommunicationDetails();
+        cd5.setLinkId("cd5");
+        cd5.setBusinessTransaction("testapp");
+        cd5.setTimestamp(1600);
+        cd5.setLatency(500);
+        cd5.setSource(EndpointUtil.encodeEndpoint(null, "in2"));
+        cd5.setTarget(EndpointUtil.encodeEndpoint(null, "out2.1"));
+        cds.add(cd5);
+
+        try {
+            analytics.storeCommunicationDetails(null, cds);
+        } catch (Exception e) {
+            fail("Failed to store: " + e);
+        }
+
+        Criteria criteria = new Criteria();
+        criteria.setStartTime(1000).setEndTime(10000);
+
+        Wait.until(() -> analytics.getCommunicationSummaryStatistics(null, criteria, false).size() == 5);
+        Collection<CommunicationSummaryStatistics> stats = analytics.getCommunicationSummaryStatistics(null,
+                            criteria, false);
+
+        assertNotNull(stats);
+        assertEquals(5, stats.size());
+
+        CommunicationSummaryStatistics in1 = null;
+        CommunicationSummaryStatistics in2 = null;
+        CommunicationSummaryStatistics out1_1 = null;
+        CommunicationSummaryStatistics out1_2 = null;
+        CommunicationSummaryStatistics out2_1 = null;
+
+        for (CommunicationSummaryStatistics css : stats) {
```

Comment:
If there is defined order in `getCommunicationSummaryStatistics` please use a `List` if not maybe is cleaner to use a transformation to a map.